### PR TITLE
bugfix for memory leak

### PR DESCRIPTION
**Pull request by JoeyFaulkner**
implemented @oroulet comments on the other pull request, as much as i understood them https://github.com/JoeyFaulkner/opcua-asyncio/pull/3

### CODE REVIEW

**Comment by JoeyFaulkner** [comment]
ok i think this error is out of my understanding https://travis-ci.org/github/FreeOpcUa/opcua-asyncio/jobs/720599077#L919

File: asyncua/server/binary_server_asyncio.py [hunk 1]
```diff
@@ -136,19 +135,29 @@ async def start(self):
...
             await self._server.wait_closed()
+
+    async def _await_closing_tasks(self):
+        await asyncio.sleep(10)
+        while self.closing_tasks:
+            task = self.closing_tasks.pop()
+            try:
+                await task
+            except Exception:
+                logger.exception("Unexpected crash in BinaryServer._await_closing_tasks")
```
Comment (swamper123) [inline asyncua/server/binary_server_asyncio.py:163]: There is an empty line missing here.

**Comment by swamper123** [comment]
This test "Run[s] an example OPC-UA server. By importing xml definition and using uawrite command line, it is even possible to expose real data using this server" within 2 seconds. After that, a SIGINT is send to the server and it has to shutdown in 2 seconds.

And there something is not working well...and I can't say exactly what. I would add a broad except block beside OSError and KeyboardInterrupt into this, perhaps you may catch an error with a better description.

(Perhaps these sleeps could be async as well, but I don't think that these are causing problems.)

File: asyncua/server/binary_server_asyncio.py [hunk 2]
```diff
@@ -136,19 +135,29 @@ async def start(self):
...
             if isinstance(result, Exception):
                 self.logger.error(f"An error ocurred while closing a transport: {result}")
-        self.protocol_tasks = []
+        self.closing_tasks = []
 
         if self._server:
             self.iserver.loop.call_soon(self._server.close)
             await self._server.wait_closed()
+
+    async def _await_closing_tasks(self):
```
Comment (swamper123) [inline asyncua/server/binary_server_asyncio.py:156]: If I understand this right, this method is just called at the start of a server? Or should it be call repeatadly?

**Review by swamper123 — CHANGES_REQUESTED** [review]

**Comment by oroulet** [comment]
Travis seemsto be down. What is the issue? It looks like master on my PC is broken too. I get a strange (RuntimeError: There is no current event loop....) No ideas where it comes from..

**Comment by swamper123** [comment]
grafik

The process returns with 2 in the test.

**Comment by oroulet** [comment]
travis is broken today. We need to wait...

**Comment by oroulet** [comment]
see the other pull request

**Pull request closed without merging**